graphs/possible_bipartition.py

```diff
@@ -1,5 +1,22 @@
 # Can be used for BFS
-from collections import deque 
+from collections import deque
+# This is DFS
+def has_no_bad_cycle(cycle_counter_list, key, dislikes):
+    # exits if the key is present in list => there is a cycle
+    # AND
+    # if the len of list(path) is odd => the cycle is NOT ok
+    if len(cycle_counter_list) % 2 != 0 and cycle_counter_list[0] == key:
```

Comment:
We know that if we find a cycle with an odd number of nodes, then there is no way to split the dogs up into two groups (we cannot bipartition the graph).

When you call your helper function, if the list passed in has an odd number of dogs (we haven't added the current dog we are looking at yet to the list) and the dog we are looking at is the first element in that least, that means that means we've found an odd cycle so we should return _False_ there _is_ a bad cycle in the list.

This will prompt your main function to also return False.